middleware/instrument.go

```diff
@@ -105,6 +110,17 @@ func (i Instrument) Wrap(next http.Handler) http.Handler {
 			labelValues = append(labelValues, tenantID)
 			instrument.ObserveWithExemplar(r.Context(), i.PerTenantDuration.WithLabelValues(labelValues...), respMetrics.Duration.Seconds())
 		}
+		if i.SlowRequestCutoff > 0 && respMetrics.Duration > i.SlowRequestCutoff {
+			parts := strings.Split(w.Header().Get("Server-Timing"), ", ")
```

Comment:
can you add some tests for this parsing? I see that there is no `instrument_test.go`, but I think how we handle some edge cases especially when parsing these headers is important to put in a test.

Since here the output is in Prometheus metrics, it's may not immediately obvious how to write tests against them (it def. wasn't to me when I started). Here's an example which asserts on the metrics the code generates.

https://github.com/grafana/dskit/blob/90d7ee0bed7c54bbc7f9792f1f1e687073b919b9/log/ratelimit_test.go#L17-L37